### DISCUSSION TITLE
Spring boot micrometer issue

### DISCUSSION THREAD

**Discussion by artemio77** [open-telemetry/opentelemetry-java-instrumentation#5758] [Q&A] [answered]
Hi guys. I am try to publish spring boot micrometer metrics via OTEL java agent. But it does not send micrometer metrics. I find that Metrics.global contains two-meter registries. 

Metrics.globalRegistry.getRegistries(); = {Collections$UnmodifiableSet@12375}  size = 2
 0 = {PrometheusMeterRegistry@12377} 
 1 = {OpenTelemetryMeterRegistry@12378}  

I think that issue that @AutoConfigureBefore(CompositeMeterRegistryAutoConfiguration.class) this configuration not work as expect.

Metrics from http://localhost:8080/actuator/metrics not present in logs. Only present OTEL custom metrics like runtime.jvm.memory.area

Example project: https://github.com/artemio77/otel-spring-boot-micrometer

OTEL agent version: 1.12.1

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @artemio77 ,
Your issues seems very similar to #5536, which was fixed, but not released yet. Can you test your application using the [latest javaagent snapshot](https://oss.sonatype.org/service/local/repositories/snapshots/content/io/opentelemetry/javaagent/opentelemetry-javaagent/1.13.0-SNAPSHOT/opentelemetry-javaagent-1.13.0-20220406.042716-113.jar)?

**Comment by artemio77** [comment]
Oh with this java agent this work as expected. Can you say when this fix will be released?

**Reply by mateuszrzeszutek** [reply]
Hmm, probably some time during next week.

**Reply by artemio77** [reply]
Perfect. Thank you!
